MiniApp/Classes/core/Display/MiniAppAdDisplayer.swift

```diff
@@ -0,0 +1,61 @@
+import Foundation
+
+/// This class is the parent class meant to be overriden when implementing ads SDK
+/// - Tag: MiniAppAdDisplayer
+internal class MiniAppAdDisplayer: NSObject, MiniAppAdDisplayDelegate {
+    #if RMA_SDK_ADMOB
+    static let shared = AdMobDisplayer()
+    #else
+    static let shared = MiniAppAdDisplayer()
+    #endif
+
+    weak var delegate: MiniAppAdDisplayDelegate?
+    var onInterstitialClosed: [String: (Result<Void, Error>) -> Void] = [:]
+    var onRewardedClosed: [String: (Result<MiniAppReward, Error>) -> Void] = [:]
+    var rewards: [String: MiniAppReward] = [:]
+
+    override init() {
+        super.init()
+        initFramework()
+    }
+
+    func initFramework() {
+    }
+
+    func loadRequestedAd(forParams params: RequestParameters?, onLoaded: @escaping (Result<Void, Error>) -> Void) {
+        guard let delegate = delegate, let params = params, let adTypeRaw = params.adType, let adType = MiniAppAdType(rawValue: adTypeRaw), let adId = params.adUnitId else {
```

Comment:
I think we should have a separate error message for the case that the host app didn't implement the delegate: i.e. the host app doesn't support displaying ads.